registry/nacos/service_discovery.go

```diff
@@ -210,8 +211,27 @@ func (n *nacosServiceDiscovery) GetRequestInstances(serviceNames []string, offse
 	return res
 }
 
+func (n *nacosServiceDiscovery) registerInstanceListener(listener registry.ServiceInstancesChangedListener) {
+	for _, t := range listener.GetServiceNames().Values() {
```

Comment:
it's not thread safe here. Should we think about `race condition`?

Reply:
It need to be synchronized.